handler/handler.go

```diff
@@ -261,21 +263,51 @@ func createCSVRow(dims []cantabular.Dimension, index, count int) []string {
 }
 
 // UploadCSVFile uploads the provided file content to AWS S3
-func (h *InstanceComplete) UploadCSVFile(ctx context.Context, instanceID string, file io.Reader) (string, error) {
+func (h *InstanceComplete) UploadCSVFile(ctx context.Context, instanceID string, file io.Reader, isPublished bool) (string, error) {
 	if instanceID == "" {
 		return "", errors.New("empty instance id not allowed")
 	}
 	if file == nil {
 		return "", errors.New("no file content has been provided")
 	}
 
+	// As the code is now it is assumed that the file is always published
+	if isPublished {
+		bucketName := h.s3.BucketName()
```

Comment:
These two lines (276, 277) can be moved to before the comment on line 274
which should allow duplicate lines 303 and 304 to be removed.

Reply:
Done